admin/templates/project/streamlit.yaml

```diff
@@ -0,0 +1,82 @@
+#!jinja2: lstrip_blocks: True
+{#
+    REQUIRED APPLICATION VARIABLES
+    ------------------------------
+    image : Docker image path
+    port  : Container port which public HTTP service runs on
+    replicas: how many instances of the application to run
+#}
+---
+
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: streamlit-app
+  namespace: {{ inventory_hostname }}
+  labels:
+    app: {{ inventory_hostname }}
+spec:
+  replicas: {{ app.replicas }}
+  strategy:
+    type: Recreate
+  selector:
+    matchLabels:
+      app: {{ inventory_hostname }}
+  template:
+    metadata:
+      labels:
+        app: {{ inventory_hostname }}
+    spec:
+      containers:
+      - image: {{ app.image }}
+        name: {{ inventory_hostname }}
+        ports:
+        - containerPort: {{ app.port }}
+          name: http
+          protocol: TCP
+      imagePullSecrets:
+      - name: regcred
+
+---
+
+apiVersion: v1
+kind: Service
+metadata:
+  name: streamlit-app
+  namespace: {{ inventory_hostname }}
+  labels:
+    app: {{ inventory_hostname }}
+spec:
+  selector:
+    app: {{ inventory_hostname }}
+  ports:
+  - name: http
+    port: 80
+    protocol: TCP
+    targetPort: {{ app.port }}
+
+---
+
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: streamlit-app
+  namespace: {{ inventory_hostname }}
+  labels:
+    app: {{ inventory_hostname }}
+  annotations:
+    certmanager.k8s.io/cluster-issuer: letsencrypt-prod
+    nginx.ingress.kubernetes.io/proxy-body-size: "0"
+spec:
+  tls:
+  - hosts:
+    - {{ inventory_hostname }}.beta-cluster.phl.io
+    secretName: tls-secret
+  rules:
+  - host: {{ inventory_hostname }}.beta-cluster.phl.io
+    http:
+      paths:
+      - path: /
+        backend:
```

Comment:
I haven't managed Ingress much, and I am unsure of this line, and the following one, which I would kind of expect to be `{{ inventory_hostname }}`, if it weren't for symmetry with `public-http` template